save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt

```diff
@@ -0,0 +1,30 @@
+package org.cqfn.save.backend.controllers
+
+import org.cqfn.save.backend.service.TestExecutionService
+import org.cqfn.save.entities.TestExecution
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+
+/**
+ * Controller to work with test execution
+ *
+ * @param testExecutionService service for test execution
+ */
+@RestController
+class TestExecutionController(private val testExecutionService: TestExecutionService) {
+    /**
+     * @param testExecutions list of test executions
+     * @return response
+     */
+    @PostMapping(value = ["/saveTestResult"])
+    fun saveTestResult(@RequestBody testExecutions: List<TestExecution>): ResponseEntity<String> {
+        try {
+            testExecutionService.saveTestResult(testExecutions)
+        } catch (exception: IllegalArgumentException) {
```

Comment:
Why specifically `IllegalArgumentException`? If you mean `save` in repository, than it throws IAE only if supplied element is null, but if it is null, then `Optional.get` would've thrown NSEE earlier.

Reply:
No, baceuse method `findById` can throw `IllegalArgumentException`

Reply:
Docs also state that it does so only if supplied id is null, if element is not present in the DB, it returns `Optional.empty`. I think IAE is too specific and rare to catch here.